diff --git a/hailort/libhailort/src/vdma/driver/os/windows/driver_os_specific.cpp b/hailort/libhailort/src/vdma/driver/os/windows/driver_os_specific.cpp
index 69d4298..3d3d5c4 100644
--- a/hailort/libhailort/src/vdma/driver/os/windows/driver_os_specific.cpp
+++ b/hailort/libhailort/src/vdma/driver/os/windows/driver_os_specific.cpp
@@ -309,12 +309,14 @@ COMPATIBLE_PARAM_CAST(hailo_write_action_list_params, WriteActionListParams)
 COMPATIBLE_PARAM_CAST(hailo_vdma_cancel_prepared_transfer_params, DescListResetParams)
 
 // Special handle for nullptr_t. This case occurs when there is no parameters passed.
+template<>
 tCompatibleHailoIoctlData WindowsIoctlParamCast<nullptr_t>::to_compatible(nullptr_t data)
 {
     (void) data;
     return tCompatibleHailoIoctlData{};
 }
 
+template<>
 void WindowsIoctlParamCast<nullptr_t>::from_compatible(const tCompatibleHailoIoctlData &compatible, nullptr_t data)
 {
     (void) compatible;
